testing/test_session.py

```diff
@@ -253,3 +256,34 @@ def pytest_sessionfinish():
     """)
     res = testdir.runpytest("--collect-only")
     assert res.ret == EXIT_NOTESTSCOLLECTED
+
+
+@pytest.mark.parametrize("path", ["root", "{relative}/root", "{environment}/root"])
+def test_rootdir_option_arg(testdir, path):
+    if 'relative' in path:
```

Comment:
You can simplify the setup because applying a `format()` call to a string without replacements will yield the original string:

```python
def test_rootdir_option_arg(testdir, path):
    path = path.format(relative=str(testdir.tmpdir), 
                       environment='$PY_ROOTDIR_PATH')
```